src/lib/radio/radio.ts

```diff
@@ -553,16 +558,25 @@ export class MatRadioButton extends _MatRadioButtonMixinBase
     // emit its event object to the `change` output.
     event.stopPropagation();
 
-    const groupValueChanged = this.radioGroup && this.value !== this.radioGroup.value;
+    if (this.radioGroup) {
+      this.updateGroupValue(true);
+    }
+
     this.checked = true;
     this._emitChangeEvent();
+  }
 
-    if (this.radioGroup) {
-      this.radioGroup._controlValueAccessorChangeFn(this.value);
-      this.radioGroup._touch();
-      if (groupValueChanged) {
+  updateGroupValue(isUserInput: boolean = false) {
```

Comment:
This method should be underscore prefixed and marked as private.  It also could use a comment documenting what it does.